src/core/net/dnssd_server.cpp

```diff
@@ -488,17 +488,25 @@ Error Server::AppendTxtRecord(Message &         aMessage,
                               uint32_t          aTtl,
                               NameCompressInfo &aCompressInfo)
 {
-    Error     error = kErrorNone;
-    TxtRecord txtRecord;
+    Error         error = kErrorNone;
+    TxtRecord     txtRecord;
+    const uint8_t kEmptyTxt = '\0';
```

Comment:
nit:
```suggestion
    const uint8_t kEmptyTxt = 0;
```

Because we are appending zero length but not null char, right?